packages/Geant4GM/source/volumes/Factory.cxx

```diff
@@ -422,17 +423,61 @@ void Geant4GM::Factory::ImportDaughters(G4LogicalVolume* lv)
     std::cout << "ImportDaughters for " << lv->GetName() << std::endl;
   }
 
-  for (size_t i = 0; i < lv->GetNoDaughters(); i++) {
+  // For parameterised volumes, the daugthers might change in solids and
+  // materials. Geant4 internally can not account for this while looping over
+  // daugthers because the logical volume is always the same. During
+  // parameterisation, only the solid and material is replaced. The actual
+  // logical volume pointer stays the same. To overcome this issue, we introduce
+  // a map of solids and materials to determine if we need a new volume to
+  // describe certain parts in parameterised volumes. All non parameterised
+  // volumes should be unaffected sind there is no change in solid / material
 
-    G4LogicalVolume* dLV = lv->GetDaughter(i)->GetLogicalVolume();
+  std::map<std::pair<G4VSolid*, G4Material*>, int> LocalSolidMaterialStore;
 
+  for (int i = 0; i < lv->GetNoDaughters(); i++) {
```

Comment:
The type in the loop should be `size_t` as `G4LogicalVolume::GetNoDaughters()` returns thjis type